src/deploy/functions/deploy.ts

```diff
@@ -63,11 +62,12 @@ export async function deploy(
   try {
     const want = payload.functions!.backend;
     const uploads: Promise<void>[] = [];
-    if (backend.allEndpoints(want).some((endpoint) => endpoint.platform === "gcfv1")) {
-      uploads.push(uploadSourceV1(context));
-    }
 
     for (const region of Object.keys(want.endpoints)) {
+      if (backend.allEndpoints(want).some((endpoint) => endpoint.platform === "gcfv1")) {
```

Comment:
This doesn't do what you think it does. This will upload to every region if any region has GCF v1.

Reply:
Which is what I want (though I can now see that we can shortcut the uploading if we've uploaded source to a region already). e.g.

```
export const fn-us = functions.region('us-central1').https...
export const fn-eu = functoins.region('europe-west1').https... 
```

Then don't we want to deploy the source to both us-central and europe-west?

Reply:
If you have a v1 function in US-central1 and a v2 function in us-west1 this will upload v1 source to both regions. I think you want:

```
if (want.endpoints[region].some(endpoint => endpoint.platform === "gcfv1"))
```